### ISSUE TITLE
Make tests/examples start

### ISSUE THREAD

**Issue by dmik** [bitwiseworks/qtwebengine-os2#4]
A continuation of #1 of some sort.

I've selected a simple unit test to check basic QtWebEngine functionality: `tests/auto/core/qtwebenginecookiestore`. First, it was spitting too many `[warn] poll: Invalid argument` messages because of `pipe` used instead of `socketpair`. After changing that the warnings go away and I now get this:
```
[14 Jul 2020 03:55:53, qt5d sh -c tst_qwebenginecookiestore.exe]

[3395:10170368:0714/035555.354000:ERROR:network_change_notifier.cc(224)] Not implemented reached in static net::NetworkChangeNotifier* net::NetworkChangeNotifier::Create()
[3395:537275104:0714/035555.473000:ERROR:network_change_notifier.cc(224)] Not implemented reached in static net::NetworkChangeNotifier* net::NetworkChangeNotifier::Create()
[3395:10170368:0714/035555.489000:ERROR:process_posix.cc(309)] Not implemented reached in base::Time base::Process::CreationTime() const
[3395:10170368:0714/035555.536000:ERROR:create_video_capture_device_factory.cc(62)] Not implemented reached in std::unique_ptr<media::VideoCaptureDeviceFactory> media::{anonymous}::CreatePlatformSpecificVideoCaptureDeviceFactory(scoped_refptr<base::SingleThreadTaskRunner>)
********* Start testing of tst_QWebEngineCookieStore *********
Config: Using QtTest library 5.13.1, Qt 5.13.1 (i386-little_endian-ilp32 shared (dynamic) release build; by GCC 9.2.0 20190812 (OS/2 RPM build 9.2.0-1))
PASS   : tst_QWebEngineCookieStore::initTestCase()
[3395:538802688:0714/035556.594000:ERROR:child_process_launcher_helper_os2.cc(22)] Not implemented reached in void content::internal::ChildProcessLauncherHelper::SetProcessPriorityOnLauncherThread(base::Process, const content::ChildProcessLauncherPriority&)
[3395:10170368:0714/035556.606000:ERROR:process_posix.cc(309)] Not implemented reached in base::Time base::Process::CreationTime() const
[0714/035556.592000:ERROR:debugger_posix.cc(176)] Not implemented reached in bool base::debug::BeingDebugged()
[3397:10170304:0714/035556.610000:ERROR:cpu_info.cc(46)] No function to get number of cores
[3397:10170304:0714/035556.624000:ERROR:broker_posix.cc(40)] Recvmsg error: Socket operation on non-socket (38)
[3397:10170304:0714/035556.643000:FATAL:page_allocator_internals_posix.h(131)] Check failed: 0 == mprotect(address, length, GetAccessFlags(accessibility)) (0 vs. -1)
[3397:10170304:0714/035556.643000:ERROR:debugger_posix.cc(176)] Not implemented reached in bool base::debug::BeingDebugged()
Received signal 6
[end of stack trace]
Calling _exit(1). Core file will not be generated.
013f1bc7 01 ff 0000 Asrt: Assertion Failed!!!
013f1bc7 01 ff 0000 Asrt: Function: <NULL>
013f1bc7 01 ff 0000 Asrt: File:     D:/Coding/libcx/master/src/mmap/mmap.c
013f1bc7 01 ff 0000 Asrt: Line:     1257
013f1bc7 01 ff 0000 Asrt: Expr:     arc == NO_ERROR
013f1bc7 01 ff 0000 Asrt: 487

===== LIBCx resource usage =====
Reserved memory size:  2097152 bytes
Committed memory size: 65536 bytes
Heap size total:       65120 bytes
Heap size used now:    10811 bytes
ProcDesc structs used now:       11
FileDesc structs used now:       40
SharedFileDesc structs used now: 26
===== LIBCx global mutex owner =====
mutex handle: 800100af
owner PID:    0d45 (3397)
owner TID:    1
===== LIBCx stats end =====
013f1e52 0c ff 0000 Asrt: Assertion Failed!!!
013f1e52 0c ff 0000 Asrt: Function: <NULL>
013f1e52 0c ff 0000 Asrt: File:     D:/Coding/libcx/master/src/shared.c
013f1e52 0c ff 0000 Asrt: Line:     550
013f1e52 0c ff 0000 Asrt: Expr:     arc == NO_ERROR
013f1e52 0c ff 0000 Asrt: 105

===== LIBCx resource usage =====
Reserved memory size:  2097152 bytes
Committed memory size: 65536 bytes
Heap size total:       65120 bytes
Heap size used now:    10751 bytes
ProcDesc structs used now:       10
FileDesc structs used now:       40
SharedFileDesc structs used now: 26
===== LIBCx stats end =====
Creating 0D43_0C.TRP
*** SPAWN 1 [D:\Coding\qt5\qt5-dev-build\qtbase\libexec\QtWebEngineProcess.exe][--type=renderer][(null)]
*** SPAWN 1 rc 3397 errno 0

[14 Jul 2020 03:55:57, exit code 0, took 4.01s]
```
I.e. it successfully starts the browser process (`QtWebEngineProcess.exe`) but then things go wild. A bunch of places to look at. The bad thing is that it screws up LIBCx so I have to reboot each time... Will look into that too. Something to do with mmap support.

**Comment by dmik** [comment]
Memory allocation problems (and crashes) have gone, now I see some communication problems between the parent and the child ("renderer") process again. Sometimes it spits
```
[warn] poll: Bad address
``` 
Sometimes just hangs until the test case times out. Needs more debugging into the messaging susbsystem.

**Comment by dmik** [comment]
I'm also getting this output among othter things:
```
[30357:10170432:0721/184950.951000:ERROR:network_change_notifier.cc(224)] Not implemented reached in static net::NetworkChangeNotifier* net::NetworkChangeNotifier::Create()
[30357:537275104:0721/184951.070000:ERROR:network_change_notifier.cc(224)] Not implemented reached in static net::NetworkChangeNotifier* net::NetworkChangeNotifier::Create()
[30357:10170432:0721/184951.084000:ERROR:process_posix.cc(309)] Not implemented reached in base::Time base::Process::CreationTime() const
[30357:10170432:0721/184951.127000:ERROR:create_video_capture_device_factory.cc(62)] Not implemented reached in std::unique_ptr<media::VideoCaptureDeviceFactory> media::{anonymous}::CreatePlatformSpecificVideoCaptureDeviceFactory(scoped_refptr<base::SingleThreadTaskRunner>)
********* Start testing of tst_QWebEngineCookieStore *********
Config: Using QtTest library 5.13.1, Qt 5.13.1 (i386-little_endian-ilp32 shared (dynamic) release build; by GCC 9.2.0 20190812 (OS/2 RPM build 9.2.0-1))
PASS   : tst_QWebEngineCookieStore::initTestCase()
*** SPAWN 1 [D:\Coding\qt5\qt5-dev-build\qtbase\libexec\QtWebEngineProcess.exe][--type=renderer][(null)]
*** SPAWN 1 rc 30360 errno 22
[30357:538802336:0721/184952.380000:ERROR:child_process_launcher_helper_os2.cc(22)] Not implemented reached in void content::internal::ChildProcessLauncherHelper::SetProcessPriorityOnLauncherThread(base::Process, const content::ChildProcessLauncherPriority&)
[30357:10170432:0721/184952.394000:ERROR:process_posix.cc(309)] Not implemented reached in base::Time base::Process::CreationTime() const
[0721/184952.117000:ERROR:debugger_posix.cc(176)] Not implemented reached in bool base::debug::BeingDebugged()
[30360:10170368:0721/184952.135000:ERROR:cpu_info.cc(46)] No function to get number of cores
[30360:10170368:0721/184952.148000:ERROR:broker_posix.cc(40)] Recvmsg error: Socket operation on non-socket (38)
FAIL!  : tst_QWebEngineCookieStore::cookieSignals() Compared values are not the same
   Actual   (((loadSpy.count()))): 0
   Expected (1)                  : 1
   Loc: [D:/Coding/qt5/qt5/qtwebengine/tests/auto/core/qwebenginecookiestore/tst_qwebenginecookiestore.cpp(98)]
[30357:538802336:0721/185007.802000:ERROR:child_process_launcher_helper_os2.cc(22)] Not implemented reached in void content::internal::ChildProcessLauncherHelper::SetProcessPriorityOnLauncherThread(base::Process, const content::ChildProcessLauncherPriority&)
FAIL!  : tst_QWebEngineCookieStore::setAndDeleteCookie() Compared values are not the same
   Actual   (((cookieAddedSpy.count()))): 0
   Expected (1)                         : 1
   Loc: [D:/Coding/qt5/qt5/qtwebengine/tests/auto/core/qwebenginecookiestore/tst_qwebenginecookiestore.cpp(132)]
FAIL!  : tst_QWebEngineCookieStore::batchCookieTasks() Compared values are not the same
   Actual   (((cookieAddedSpy.count()))): 0
   Expected (1)                         : 1
   Loc: [D:/Coding/qt5/qt5/qtwebengine/tests/auto/core/qwebenginecookiestore/tst_qwebenginecookiestore.cpp(169)]
```
Most ERROR messages are harmless, but the one in `broker_posix.cc` looks like a bad one. Seems that the mojo transport isn't working well yet. Mojo is a Google own IPC implementation (see [here](https://chromium.googlesource.com/chromium/src.git/+/master/mojo/README.md)).

**Comment by dmik** [comment]
With some fixes I went further, now it crashes in Qt5Core:
```
 Trap  ->  1D4B269C   QT5CORE   0001:0026269C  qresource.cpp#1401 __ZN19QResourceFileEngine4readEPcx + 8C 0001:00262610 (D:\Coding\qt5\qt5\qtbase\src\corelib\io\qresource.cpp)
 036FF7E4  1D488E3D   QT5CORE   0001:00238E3D  qfiledevice.cpp#463 __ZN11QFileDevice8readDataEPcx + 7D 0001:00238DC0 (D:\Coding\qt5\qt5\qtbase\src\corelib\io\qfiledevice.cpp)
 036FF834  1D495D2E   QT5CORE   0001:00245D2E  qiodevice.cpp#1114 __ZN16QIODevicePrivate4readEPcxb + 1DE 0001:00245B50 (D:\Coding\qt5\qt5\qtbase\src\corelib\io\qiodevice.cpp)
 036FF8C4  1D4964ED   QT5CORE   0001:002464ED  qiodevice.cpp#1063 __ZN9QIODevice4readEPcx + 18D 0001:00246360 (D:\Coding\qt5\qt5\qtbase\src\corelib\io\qiodevice.cpp)
 036FF914  0C2863E9   QT5WEBC   0001:000563E9   __ZN15QtWebEngineCore19URLRequestCustomJob11ReadRawDataEPN3net8IOBufferEi + 49 0001:000563A0 (D:\Coding\qt5\qt5\qtwebengine\src\core\net\url_request_custom_job.cpp)
 20297620  20299A40   *Unknown*
Lost Stack chain - new EBP below previous
```
I guess it tries to read some test URL from a resource file which is missing on OS/2.

Another issue, besides the above one, is that I'm also suffering due to a couple of EXCEPTQ limitations. The first one is inability to properly parse optimized GCC stack frames which usually ends up in a premature stack trace failure, just like above. Another problem is that if two exceptions on two different threads happen about the same time, EXCEPTQ only generates one of them (and not always the one I need). For instance, it spits this:

```
Creating 5970_0E.TRP
** Secondary exception C0000005 occurred in TID 5 at cs:eip 005B:1D4B269C **
** Exceptq report suppressed - exceptq in use by TID 14 **
[22896:5:0729/181558.495000:ERROR:debugger_posix.cc(176)] Not implemented reached in bool base::debug::BeingDebugged()
Received signal 11 SEGV_ACCERR 0000b5bb8000
[end of stack trace]
Calling _exit(1). Core file will not be generated.
```

I.e. I'm getting a report for TID 14 which is quite useless because it's some consequence of another one and because of the above mentioned problem with the premature stack trace end. The only thing I get is this:
```
Trap  ->  17B40008   QT5WEBC   0004:00000008
```

And the real trap is TID 5 but I'm not getting a .TRP for it...

@StevenLevine, if you follow me here, I'd like to know how hard it would be to fix both issues in EXCEPTQ.

**Comment by dmik** [comment]
I'm getting another crash now:
```
______________________________________________________________________

 Exception C0000005 - Access Violation
______________________________________________________________________

 Process:  D:\CODING\QT5\QT5-DEV-BUILD\QTWEBENGINE\TESTS\AUTO\CORE\QWEBENGINECOOKIESTORE\DEBUG\TST_QWEBENGINECOOKIESTORE.EXE (08/09/2020 13:12:17 448,947)
 PID:      38C6 (14534)
 TID:      0D (13)
 Priority: 200

 Filename: D:\CODING\QT5\QT5-DEV-BUILD\QTBASE\LIB\QT5WEBCD.DLL (08/10/2020 17:59:19 354,093,941)
 Address:  005B:B860424C (0001:0599424C)
 Cause:    Attempted to read from B1C969DD
           (uncommitted memory allocated by LIBCX0)

______________________________________________________________________

 Failing Instruction
______________________________________________________________________

 B860423F  JMP   0xb860424c       (eb 0b)
 B8604241  LEA   ESI, [ESI+0x0]   (8db426 00000000)
 B8604248  CMP   EAX, EDX         (39d0)
 B860424A  JAE   0xb8604258       (73 0c)
 B860424C >MOVSX ESI, BYTE [EAX]  (0fbe30)
 B860424F  ADD   EAX, 0x1         (83c0 01)
 B8604252  OR    EBX, ESI         (09f3)
 B8604254  TEST  AL, 0x3          (a8 03)

______________________________________________________________________

 Registers
______________________________________________________________________

 EAX : B1C969DD   EBX  : 00000000   ECX : B1C969DD   EDX  : B1C969F9
 ESI : 0447C2F4   EDI  : 20049B80
 ESP : 0447C1DC   EBP  : 0447C254   EIP : B860424C   EFLG : 00010246
 CS  : 005B       CSLIM: FFFFFFFF   SS  : 0053       SSLIM: FFFFFFFF

 EAX : uncommitted memory allocated by LIBCX0
 EBX : not a valid address
 ECX : uncommitted memory allocated by LIBCX0
 EDX : uncommitted memory allocated by LIBCX0
 ESI : read/write memory on this thread's stack
 EDI : read/write memory allocated by LIBCN0

______________________________________________________________________

 Stack Info for Thread 0D
______________________________________________________________________

   Size       Base        ESP         Max         Top
 00200000   04480000 -> 0447C1DC -> 04478000 -> 04280000

______________________________________________________________________

 Call Stack
______________________________________________________________________

   EBP     Address    Module     Obj:Offset    Nearest Public Symbol
 --------  ---------  --------  -------------  -----------------------
 Trap  ->  B860424C   QT5WEBCD  0001:0599424C   __ZNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEE12_M_constructIPKcEEvT_S8_St20forward_iterator_tag$w$vuBQdW1xLM7Codx60 + CC 0001:05994180 (base_jumbo_19.o)

 0447C254  B86045F2   QT5WEBCD  0001:059945F2   __ZN4base11UTF8ToUTF16ENS_16BasicStringPieceINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEE + 32 0001:059945C0 (base_jumbo_19.o)

 0447C274  B87A8FEE   QT5WEBCD  0001:05B38FEE   __ZN2ui14ResourceBundle22GetLocalizedStringImplB5cxx11Ei + 2FE 0001:05B38CF0 (base_jumbo_4.o)

 0447C414  B87A961E   QT5WEBCD  0001:05B3961E   __ZN2ui14ResourceBundle18GetLocalizedStringB5cxx11Ei + 13E 0001:05B394E0 (base_jumbo_4.o)

 0447C554  B86D4262   QT5WEBCD  0001:05A64262   __ZN9l10n_util14GetStringUTF16B5cxx11Ei + 22 0001:05A64240 (base_jumbo_2.o)

 0447C574  B8A6947E   QT5WEBCD  0001:05DF947E   __ZN10error_page14LocalizedError10GetStringsEiRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERK4GURLbbbbNS0_36OfflineContentOnNetErrorFeatureStateEbS8_St10unique_ptrINS_15ErrorPageParamsESt14default_deleteISE_EEPN4base15!_aYIq24 + 2CE 0001:05DF91B0 (localized_error.o)

 0447C7B4  B2D1B0EE   QT5WEBCD  0001:000AB0EE   __ZN15QtWebEngineCore23ContentRendererClientQt33GetNavigationErrorStringsInternalEPN7content11RenderFrameERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKN10error_page5ErrorEPS9_ + 12E 0001:000AAFC0 (D:\Coding\qt5\qt5\qtwebengine\src\core\renderer\content_renderer_client_qt.cpp)

 0447C944  B2D1B4F9   QT5WEBCD  0001:000AB4F9   __ZN15QtWebEngineCore23ContentRendererClientQt16PrepareErrorPageEPN7content11RenderFrameERKN5blink11WebURLErrorERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEbPSD_ + C9 0001:000AB430 (D:\Coding\qt5\qt5\qtwebengine\src\core\renderer\content_renderer_client_qt.cpp)

 0447CAA4  B8B1853D   QT5WEBCD  0001:05EA853D   __ZN7content15RenderFrameImpl23LoadNavigationErrorPageEPN5blink17WebDocumentLoaderERKNS1_11WebURLErrorERKN4base8OptionalINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEbb + 5BD 0001:05EA7F80 (render_frame_impl.o)

 0447CC54  B8B3B521   QT5WEBCD  0001:05ECB521   __ZN7content15RenderFrameImpl22DidFailProvisionalLoadERKN5blink11WebURLErrorENS1_20WebHistoryCommitTypeE + 211 0001:05ECB310 (render_frame_impl.o)

 0447CD44  B8CA145D   QT5WEBCD  0001:0603145D   __ZN5blink17WebLocalFrameImpl7DidFailERKNS_13ResourceErrorEbNS_20WebHistoryCommitTypeE + 83 0001:060313DA (frame_jumbo_10.o)
```

The code is trying to locate some locale-specific string in a resource pack file that it maps with `mmap`. For some reason, the LIBCx exception handler is called only once to read the file in. And the other time it is not called (at least I don't see it in LIBCx logs), memory is not committed and file contents is not read into it. Hence, access violation.

The strange thing is that I don't see any activity of the thread that performs this memory access in LIBCx log. Perhaps, this thread is somehow started bypassing LIBCx overrides so no exception handler is installed...

**Comment by dmik** [comment]
Ok, I know what it is, it's pretty weird. I was using a debug build LIBCx (LIBPATHSTRICT=T) but QtWebEngine also uses the pthread DLL which is not under effect of LIBPATHSTRICT=T and therefore it uses LIBCx which is not under its effect either. Therefore the exception handler from a wrong DLL was installed (and then called) which knows nothing about the given mapping (as it uses a separate shared memory block to store its global data). Using one copy of LIBCx solves this dummy issue.

This results in a test case finished but with some test failures, here they are:
```
FAIL!  : tst_QWebEngineCookieStore::cookieSignals() 'success.toBool()' returned FALSE. ()
   Loc: [D:/Coding/qt5/qt5/qtwebengine/tests/auto/core/qwebenginecookiestore/tst_qwebenginecookiestore.cpp(105)]
FAIL!  : tst_QWebEngineCookieStore::setAndDeleteCookie() 'success.toBool()' returned FALSE. ()
   Loc: [D:/Coding/qt5/qt5/qtwebengine/tests/auto/core/qwebenginecookiestore/tst_qwebenginecookiestore.cpp(150)]
FAIL!  : tst_QWebEngineCookieStore::batchCookieTasks() 'success.toBool()' returned FALSE. ()
   Loc: [D:/Coding/qt5/qt5/qtwebengine/tests/auto/core/qwebenginecookiestore/tst_qwebenginecookiestore.cpp(186)]
FAIL!  : tst_QWebEngineCookieStore::basicFilter() 'loadSpy.takeFirst().takeFirst().toBool()' returned FALSE. ()
   Loc: [D:/Coding/qt5/qt5/qtwebengine/tests/auto/core/qwebenginecookiestore/tst_qwebenginecookiestore.cpp(215)]
FAIL!  : tst_QWebEngineCookieStore::html5featureFilter() 'loadSpy.takeFirst().takeFirst().toBool()' returned FALSE. ()
   Loc: [D:/Coding/qt5/qt5/qtwebengine/tests/auto/core/qwebenginecookiestore/tst_qwebenginecookiestore.cpp(245)]
```
Not sure why they fail yet.
